tests/test_self.py

```diff
@@ -258,81 +255,6 @@ def test_py3k_jobs_option(self):
             code=rc_code,
         )
 
-    @pytest.mark.skipif(sys.version_info[0] > 2, reason="Requires the --py3k flag.")
```

Comment:
They fail on Python 3:

```
=================================== FAILURES ===================================
_______________ TestRunTC.test_py3k_commutative_with_errors_only _______________
self = <test_self.TestRunTC object at 0x7f858d028588>
    def test_py3k_commutative_with_errors_only(self):
    
        # Test what gets emitted with -E only
        module = join(HERE, "regrtest_data", "py3k_error_flag.py")
        expected = textwrap.dedent(
            """
        ************* Module py3k_error_flag
        Explicit return in __init__
        """
        )
        self._test_output(
>           [module, "-E", "--msg-template='{msg}'"], expected_output=expected
        )
../tests/test_self.py:269: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <test_self.TestRunTC object at 0x7f858d028588>
args = ['/home/travis/build/hugovk/pylint/tests/regrtest_data/py3k_error_flag.py', '-E', "--msg-template='{msg}'"]
expected_output = '\n************* Module py3k_error_flag\nExplicit return in __init__\n'
    def _test_output(self, args, expected_output):
        out = StringIO()
        self._run_pylint(args, out=out)
        actual_output = self._clean_paths(out.getvalue())
        expected_output = self._clean_paths(expected_output)
>       assert expected_output.strip() in actual_output.strip()
E       AssertionError: assert '************* Module py3k_error_flag\nExplicit return in __init__' in '************* Module py3k_error_flag\ninvalid syntax (<unknown>, line 4)'
E        +  where '************* Module py3k_error_flag\nExplicit return in __init__' = <built-in method strip of str object at 0x7f858d7f1af8>()
E        +    where <built-in method strip of str object at 0x7f858d7f1af8> = '\n************* Module py3k_error_flag\nExplicit return in __init__\n'.strip
E        +  and   '************* Module py3k_error_flag\ninvalid syntax (<unknown>, line 4)' = <built-in method strip of str object at 0x7f8591a2d630>()
E        +    where <built-in method strip of str object at 0x7f8591a2d630> = '************* Module py3k_error_flag\ninvalid syntax (<unknown>, line 4)\n'.strip
../tests/test_self.py:135: AssertionError
_____________ TestRunTC.test_py3k_commutative_with_config_disable ______________
self = <test_self.TestRunTC object at 0x7f858cf79390>
    def test_py3k_commutative_with_config_disable(self):
        module = join(HERE, "regrtest_data", "py3k_errors_and_warnings.py")
        rcfile = join(HERE, "regrtest_data", "py3k-disabled.rc")
        cmd = [module, "--msg-template='{msg}'", "--reports=n"]
    
        expected = textwrap.dedent(
            """
        ************* Module py3k_errors_and_warnings
        import missing `from __future__ import absolute_import`
        Use raise ErrorClass(args) instead of raise ErrorClass, args.
        Calling a dict.iter*() method
        print statement used
        """
        )
>       self._test_output(cmd + ["--py3k"], expected_output=expected)
../tests/test_self.py:302: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <test_self.TestRunTC object at 0x7f858cf79390>
args = ['/home/travis/build/hugovk/pylint/tests/regrtest_data/py3k_errors_and_warnings.py', "--msg-template='{msg}'", '--reports=n', '--py3k']
expected_output = '\n************* Module py3k_errors_and_warnings\nimport missing `from __future__ import absolute_import`\nUse raise ErrorClass(args) instead of raise ErrorClass, args.\nCalling a dict.iter*() method\nprint statement used\n'
    def _test_output(self, args, expected_output):
        out = StringIO()
        self._run_pylint(args, out=out)
        actual_output = self._clean_paths(out.getvalue())
        expected_output = self._clean_paths(expected_output)
>       assert expected_output.strip() in actual_output.strip()
E       AssertionError: assert '************* Module py3k_errors_and_warnings\nimport missing `from __future__ import absolute_import`\nUse raise ErrorClass(args) instead of raise ErrorClass, args.\nCalling a dict.iter*() method\nprint statement used' in ''
E        +  where '************* Module py3k_errors_and_warnings\nimport missing `from __future__ import absolute_import`\nUse raise ErrorClass(args) instead of raise ErrorClass, args.\nCalling a dict.iter*() method\nprint statement used' = <built-in method strip of str object at 0x7f858d0c67a0>()
E        +    where <built-in method strip of str object at 0x7f858d0c67a0> = '\n************* Module py3k_errors_and_warnings\nimport missing `from __future__ import absolute_import`\nUse raise ErrorClass(args) instead of raise ErrorClass, args.\nCalling a dict.iter*() method\nprint statement used\n'.strip
E        +  and   '' = <built-in method strip of str object at 0x7f85a38f1c70>()
E        +    where <built-in method strip of str object at 0x7f85a38f1c70> = ''.strip
../tests/test_self.py:135: AssertionError
==== 2 failed, 1069 passed, 56 skipped, 49 deselected in 104.89s (0:01:44) =====
```